[export] Serialize subgraphs.

zhxchen17 opened the pull request:
Differential Revision: D46865179

Deserialization part will be added in a following up PR.

larryliu0820 commented on torch/_export/serde/serialize.py:
```diff
@@ -463,11 +485,22 @@ def handle_call_function(self, node: torch.fx.Node):
+            inputs = [NamedArgument(
+                name="", # TODO(zhxchen17) This is sad, should be improved when HOO has schema arg names.
+                arg=self.serialize_input(arg),
+            ) for arg in node.args]
```
`arg` is shadowing `NamedArgument.arg`. Consider changing the variable name.

larryliu0820 reviewed (commented):
Can you add a test for this?

zhxchen17 commented:
> Can you add a test for this?

I will add a test in the deserialization diff later.

zhxchen17 committed:
[export] Serialize subgraphs. (#103901)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/103901

Differential Revision: D46865179

fbshipit-source-id: 25a78a7cea89406a9aec53e55b12d4d9c8e209eb

zhxchen17 commented:
@pytorchbot merge

zhxchen17 commented on torch/_export/serde/serialize.py:
```diff
@@ -513,18 +549,21 @@ def serialize_inputs(
-                return Argument.create(as_tensor=TensorArgument(name=str(arg.target)))
+                assert isinstance(arg.target, str)
+                with self.save_graph_state():
+                    graph = self.serialize_graph(getattr(arg.graph.owning_module, arg.target))
```
??
